assets/js/sidebar.js

```diff
@@ -0,0 +1,53 @@
+var $ = require('jquery')
+
+var body = $('body')
+var sidebar = $('.sidebar')
+
+var bodyClass = 'sidebar-open'
+var duration = 300
+var width = '300px'
+
+function closeSidebar () {
+  sidebar.animate({
+    'flex-basis': 0
+  }, duration, function () {
+    body.removeClass(bodyClass)
+  })
+}
+
+function openSidebar (immediate) {
+  var dur = duration
+  if (immediate) {
+    dur = 0
+  }
+  sidebar.animate({
+    '-webkit-flex-basis': width,
+    '-moz-flex-basis': width,
+    '-ms-flex-basis': width,
+    'flex-basis': width
+  }, 0, function () {
```

Comment:
I think that at this point you should reference the variable `dur` to indicate the duration and not a fixed value to perform the animation, right?